qiskit/circuit/library/basis_change/qft.py

```diff
@@ -0,0 +1,236 @@
+        q_2: ─┼──X──░──────────────■──────┼──────░───────■──────┼──────░─┤ H ├─■──────░───────░─
+              │     ░                     │pi/8  ░              │pi/4  ░ └───┘ │pi/2  ░ ┌───┐ ░
+        q_3: ─X─────░─────────────────────■──────░──────────────■──────░───────■──────░─┤ H ├─░─
+                    ░                            ░                     ░              ░ └───┘ ░
+
+    The inverse QFT can be obtained by calling the ``.inverse()`` method on this class.
+    The respective circuit diagram is:
+
+    .. parsed-literal::
+                    ░               ░                       ░                         ┌───┐ ░
+        q_0: ───────░───────────────░───────────────────────░──■───────■───────■──────┤ H ├─░─────X─
+                    ░               ░                 ┌───┐ ░  │       │       │-pi/2 └───┘ ░     │
+        q_1: ───────░───────────────░──■───────■──────┤ H ├─░──┼───────┼───────■────────────░──X──┼─
+                    ░         ┌───┐ ░  │       │-pi/2 └───┘ ░  │       │-pi/4               ░  │  │
+        q_2: ───────░──■──────┤ H ├─░──┼───────■────────────░──┼───────■────────────────────░──X──┼─
+              ┌───┐ ░  │-pi/2 └───┘ ░  │-pi/4               ░  │-pi/8                       ░     │
+        q_3: ─┤ H ├─░──■────────────░──■────────────────────░──■────────────────────────────░─────X─
+              └───┘ ░               ░                       ░                               ░
+
+    One method to reduce circuit depth is to implement the QFT approximately by ignoring
+    controlled-phase rotations where the angle is beneath a threshold. This is discussed
+    in more detail in https://arxiv.org/abs/quant-ph/9601018 or
+    https://arxiv.org/abs/quant-ph/0403071.
+
+    Here, this can be adjusted using the ``approximation_degree`` attribute: the smallest
+    ``approximation_degree`` rotation angles are dropped from the QFT. For instance, a QFT
+    on 5 qubits with approximation degree 2 yields (the barriers are dropped in this example):
+
+    .. parsed-literal::
+                ┌───┐
+        q_0: ─X─┤ H ├─■──────■────────────────────────────────────────────────────────────
+              │ └───┘ │pi/2  │     ┌───┐
+        q_1: ─┼───X───■──────┼─────┤ H ├─■──────■─────────────────────────────────────────
+              │   │          │pi/4 └───┘ │pi/2  │     ┌───┐
+        q_2: ─┼───┼──────────■───────────■──────┼─────┤ H ├─■──────■──────────────────────
+              │   │                             │pi/4 └───┘ │pi/2  │     ┌───┐
+        q_3: ─┼───X─────────────────────────────■───────────■──────┼─────┤ H ├─■──────────
+              │                                                    │pi/4 └───┘ │pi/2 ┌───┐
+        q_4: ─X────────────────────────────────────────────────────■───────────■─────┤ H ├
+                                                                                     └───┘
+    """
+
+    def __init__(self, *regs,
```

Comment:
can this be just `num_qubits` for consistency with the rest of the circuit library? The name of a register should not matter here, just how many qubits there are.